src/System.Text.Utf8String/System/Text/Utf8String.cs

```diff
@@ -11,29 +11,48 @@
 namespace System.Text.Utf8
 {
     [DebuggerDisplay("{ToString()}u8")]
-    public partial ref struct Utf8String
+    public partial class Utf8String
```

Comment:
Can still be a `struct`?

Reply:
The idea was that we keep it as a class. It avoids accidental boxing, especially that this type will probably implement some interfaces, e.g. IEquatable.

We will ultimately have two utf8string: fast and slow.
This one is slow. The fast one will be length + data inline, just like System.String.
See https://github.com/dotnet/corefxlab/issues/1751#issuecomment-329936409

The only reason for making it a struct (that I see) is that it would allow fast and slow string to have the same perfromance characteristics, but I am not sure this is the right tradeoff.